### PR TITLE
adding meta data support

### PR DESCRIPTION
**Pull request by jprincipe**
this PR adds support for processing meta data from the JSON payload.  For backwards compatibility I left the `sync()` method the same and added a new `syncWithMeta()` method that returns both the data and meta data at the same time.  The key for meta data is defaulted to `meta` but it can be overridden using the `setMetaKey` method.

### CODE REVIEW

**Comment by beauby** [comment]
Very useful PR indeed, although I believe the JSON API spec does not allow for a "custom meta key", so the `setMetaKey` kind of bothers me. Thoughts?

**Comment by jprincipe** [comment]
I agree based on the spec. I only added it because active_model_serializers allows for customizing the key

**Comment by beauby** [comment]
Actually AMS allows this only for the `Json` serializer, not for the `JsonApi` one. I think we can remove it for now, and if there's a strong need for it (which I doubt) we'll add it back.

File: dist/jsonapi-datastore.js [hunk 1]
```diff
@@ -220,6 +231,25 @@ var JsonApiDataStore = (function () {
...
+  }, {
+    key: "syncWithMeta",
+    value: function syncWithMeta(payload) {
+      var primary = payload.data,
+          syncRecord = this.syncRecord.bind(this);
+      if (!primary) return [];
+      if (payload.included) payload.included.map(syncRecord);
+      return {
+        data: primary.constructor === Array ? primary.map(syncRecord) : syncRecord(primary),
+        meta: this.metaKey in payload ? payload[this.metaKey] : {}
```
Comment (beauby) [inline dist/jsonapi-datastore.js]: Here I'd set meta to `null` instead of an empty hash.

**Comment by jprincipe** [comment]
Just looked at the ams source and `meta_key` is defined in the base adapter.  I see that you have a change in ams master around the meta_key.  Is that code in rc3?  

https://github.com/rails-api/active_model_serializers/blob/6018ef16c41c441aeb1621852c6f58571b80e656/lib/active_model/serializer/adapter/base.rb#L43

I tested this out locally on rc3 and sure enough if I set meta_key in the render, the meta data is returned with the custom key

**Comment by beauby** [comment]
Yeah, but this is just a byproduct of shared code between adapters. AMS does not officially support custom meta key, and I'll probably get rid of it when I'm done with deserialization and some other stuff.

**Comment by jprincipe** [comment]
@beauby I removed the meta_key from the code and defaulted the meta response to null if the `meta` key doesn't exist

File: README.md [hunk 2]
```diff
@@ -145,7 +154,7 @@ See [DOCUMENTATION.md](DOCUMENTATION.md).
 
 ## What's missing
 
-Currently, the store does not handle `links` and `meta` attributes.
+Currently, the store does not handle `links` attributes.
```
Comment (beauby) [inline README.md]: So here, we should say that it handles the top level meta property, but that it does not handle resource-level or relationship-level meta.

**Comment by beauby** [comment]
LGTM – If you could just fix the docs to specify which `meta` property it supports, that'd be great! Thanks again :+1:

**Pull request closed without merging**